raiden/transfer/state.py

```diff
@@ -131,7 +131,7 @@ def __post_init__(self) -> None:
 
 
 @dataclass
-class RouteState(State):
+class RouteState(State, MetadataValidation):
```

Comment:
Classes listed earlier in the inheritance order take precedence in the MRO, therefore mixins should generally be listed in front and "proper" base classes at the end.

`MetadataValidation` isn't a true mixin but kind of behaves like one. Therefore I'd say the order should be swapped here.
(Also `State` is a dataclass and `MetadataValidation` isn't.)

Reply:
Good point ✅